wdl-lint/src/rules/comment_whitespace.rs

```diff
@@ -0,0 +1,178 @@
+            // check preceding whitespace for two spaces
+            if let Some(prior) = comment.syntax().prev_sibling_or_token() {
+                if prior.kind() != SyntaxKind::Whitespace || prior.to_string() != "  " {
+                    // Report a diagnostic if there are not two spaces before the comment delimiter
+                    state.add(inline_preceding_whitespace(comment.span()))
+                }
+            }
+        } else {
+            // Not an in-line comment, so check indentation level
+            let ancestors = comment.syntax().parent_ancestors().collect::<Vec<_>>();
+            let expected_indentation = INDENT.repeat(ancestors.len() - 1);
+
+            if let Some(leading_whitespace) = comment.syntax().prev_sibling_or_token() {
+                let this_whitespace = leading_whitespace.to_string();
+                let this_indentation = this_whitespace
+                    .split('\n')
+                    .last()
+                    .expect("should have prior whitespace");
+                if this_indentation != expected_indentation {
+                    // Report a diagnostic if the comment is not indented properly
+                    match this_indentation.len().cmp(&expected_indentation.len()) {
+                        Ordering::Greater => state.add(excess_indentation(
+                            comment.span(),
+                            expected_indentation.len() / INDENT.len(),
+                            this_indentation.len() / INDENT.len(),
+                        )),
+                        Ordering::Less => state.add(insufficient_indentation(
+                            comment.span(),
+                            expected_indentation.len() / INDENT.len(),
+                            this_indentation.len() / INDENT.len(),
+                        )),
+                        Ordering::Equal => {}
+                    }
+                }
+            } else {
+                // If there is no prior whitespace, this comment must be at the
+                // start of the file.
+            }
+        }
+
+        // check the comment for one space following the comment delimiter
+        if let Some(delimiter) = re.captures(comment.as_str()) {
+            let d = delimiter.get(0).unwrap();
+            let rest = &comment.as_str()[d.len()..];
+            let without_spaces = rest.trim_start_matches(' ');
+
+            if !rest.is_empty() && rest.len() - without_spaces.len() != 1 {
+                // Report a diagnostic if there is not one space after the comment delimiter
+                state.add(following_whitespace(Span::new(
+                    comment.span().start(),
+                    d.len(),
+                )));
+            }
+        } else {
+            unreachable!("A comment must start with a #")
+        }
+    }
+}
+
+/// Detect is a comment is in-line or not by looking for `\n` in the prior
+/// whitespace.
+fn is_inline_comment(token: &Comment) -> bool {
```

Comment:
Should probably have a small unit test to make sure this is working appropriately.

Reply:
Like this: https://github.com/stjude-rust-labs/wdl/pull/138/files#diff-4406a4f641401d5eb6bd18bb1aaba2229755b82832da763fd464782f34e33d78R175